File: NodeJS/sitemap_scraper_requests.js
const axios = require('axios');
const fs = require('fs');
const { parseStringPromise } = require('xml2js');

const sitemapUrl = 'https://demo.nopcommerce.com/sitemap.xml';
const outputFile = 'sitemap_links.txt';

(async () => {
  try {
    const response = await axios.get(sitemapUrl);
    const xml = response.data;

    const result = await parseStringPromise(xml);
    const urls = result.urlset.url.map(entry => entry.loc[0]);

    fs.writeFileSync(outputFile, urls.join('\n'), 'utf-8');
    console.log(`Saved ${urls.length} links to ${outputFile}`);
  } catch (err) {
    console.error('Error:', err.message);
  }
})();

File: Python/sitemap_scraper_requests.py
import requests
import xml.etree.ElementTree as ET

sitemap_url = "https://demo.nopcommerce.com/sitemap.xml"
output_file = "sitemap_links.txt"

response = requests.get(sitemap_url)
response.raise_for_status()

root = ET.fromstring(response.content)
namespace = {'ns': 'http://www.sitemaps.org/schemas/sitemap/0.9'}
links = [loc.text for loc in root.findall('.//ns:loc', namespace)]

with open(output_file, 'w', encoding='utf-8') as f:
    for link in links:
        f.write(link + '\n')